Mathlib/Topology/Algebra/OpenSubgroup.lean

```diff
@@ -454,3 +454,107 @@ instance [ContinuousMul G] : Lattice (OpenNormalSubgroup G) :=
 end OpenNormalSubgroup
 
 end
+
+/-!
+## The Open Subgroup in a Clopen Neighborhood of One
+This section define `OpenSubgroupSubClopenNhdsOfOne` (`OpenNormalSubgroupSubClopenNhdsOfOne`),
+which can pick an open subgroup contained in a clopen neighborhood of `1`.
+-/
+
+open scoped Pointwise
+
+variable {G : Type*} [TopologicalSpace G]
+
+structure TopologicalAddGroup.addNegClosureNhd (T W : Set G) [AddGroup G] : Prop where
+  nhd : T ∈ 𝓝 0
+  inv : -T = T
+  isOpen : IsOpen T
+  mul : W + T ⊆ W
+
+/--For a set W, the neighborhood of `1` which is open, self inverse and satisfying `T * W ⊆ W` -/
+@[to_additive
+"For a set W, the neighborhood of `0` which is open, self negative and satisfying `T + W ⊆ W`"]
+structure TopologicalGroup.mulInvClosureNhd (T W : Set G) [Group G] : Prop where
+  nhd : T ∈ 𝓝 1
+  inv : T⁻¹ = T
+  isOpen : IsOpen T
+  mul : W * T ⊆ W
+
+namespace TopologicalGroup
+
+variable [Group G] [TopologicalGroup G] [CompactSpace G]
+
+open Set Filter
+
+@[to_additive]
+lemma exist_mul_closure_nhd {W : Set G} (WClopen : IsClopen W) : ∃ T ∈ 𝓝 (1 : G), W * T ⊆ W := by
+  apply WClopen.isClosed.isCompact.induction_on (p := fun S ↦ ∃ T ∈ 𝓝 (1 : G), S * T ⊆ W)
+    ⟨Set.univ ,by simp only [univ_mem, empty_mul, empty_subset, and_self]⟩
+    (fun _ _ huv ⟨T, hT, mem⟩ ↦ ⟨T, ⟨hT, (mul_subset_mul_right huv).trans mem⟩⟩)
+    fun U V ⟨T₁, hT₁, mem1⟩ ⟨T₂, hT₂, mem2⟩ ↦ ⟨T₁ ∩ T₂, ⟨inter_mem hT₁ hT₂, by
+      rw [union_mul]
+      exact union_subset (mul_subset_mul_left inter_subset_left |>.trans mem1)
+        (mul_subset_mul_left inter_subset_right |>.trans mem2)⟩⟩
+  intro x memW
+  have : (x, 1) ∈ (fun p ↦ p.1 * p.2) ⁻¹' W := by simp only [mem_preimage, mul_one, memW]
```

Comment:
Is there any reason to use `simp only` here? Is `simp [memW]` too slow?

Reply:
I was once told not to leave amy `simp []` and use `simp only` instead because some stability between versions. So what is the convention we usually follow? Thank you

Reply:
This is only for non-terminal simps, or slow ones. Please read https://leanprover-community.github.io/extras/simp.html.